### ISSUE TITLE
Replication Rack write/read not working

### ISSUE THREAD

**Issue by balajivenki** [Netflix/dyno#146]
With v1.5.1. If the primary node in local rack is down write or read to the corresponding node in replication rack or other rack is not working. It used to work well prior to 1.5.1. With local rack affinity changes it is regressed.

E,g. In my test case the key "ApZmMHHkWN" is part of node2 in rack1 and also node2 in rack2. I brought down dynomite(redis is still up). When I write a new value or read a value for the key the expected behavior is that since node2 in rack1 is down it should read from node2 from rack2. But I am getting the following exception.

Exception in thread "main" com.netflix.dyno.connectionpool.exception.FatalConnectionException: FatalConnectionException: [host=Host [hostname=UNKNOWN, ipAddress=UNKNOWN, port=0, rack: null, datacenter: null, status: Down], latency=0(0), attempts=1]redis.clients.jedis.exceptions.JedisDataException: ERR Peer: Peer Node is not connected
    at com.netflix.dyno.jedis.JedisConnectionFactory$JedisConnection.execute(JedisConnectionFactory.java:104)
    at com.netflix.dyno.connectionpool.impl.ConnectionPoolImpl.executeWithFailover(ConnectionPoolImpl.java:294)
    at com.netflix.dyno.jedis.DynoJedisClient.d_get(DynoJedisClient.java:340)
    at com.netflix.dyno.jedis.DynoJedisClient.get(DynoJedisClient.java:334)
    at ConfLoaded.main(ConfLoaded.java:31)
Caused by: redis.clients.jedis.exceptions.JedisDataException: ERR Peer: Peer Node is not connected
    at redis.clients.jedis.Protocol.processError(Protocol.java:127)
    at redis.clients.jedis.Protocol.process(Protocol.java:161)
    at redis.clients.jedis.Protocol.read(Protocol.java:215)
    at redis.clients.jedis.Connection.readProtocolWithCheckingBroken(Connection.java:340)
    at redis.clients.jedis.Connection.getBinaryBulkReply(Connection.java:259)
    at redis.clients.jedis.Connection.getBulkReply(Connection.java:248)
    at redis.clients.jedis.Jedis.get(Jedis.java:153)
    at com.netflix.dyno.jedis.DynoJedisClient$9.execute(DynoJedisClient.java:343)
    at com.netflix.dyno.jedis.DynoJedisClient$9.execute(DynoJedisClient.java:340)
    at com.netflix.dyno.jedis.JedisConnectionFactory$JedisConnection.execute(JedisConnectionFactory.java:85)
    ... 4 more

**Comment by ipapapa** [comment]
It looks to me that `ERR Peer: Peer Node is not connected` comes from Dynomite and is not related to Dyno. We have not made any changes from 1.5.0 to 1.5.1 other than your changes. Are you sure the same exactly use case was working before?

**Comment by balajivenki** [comment]
Hi @ipapapa ,
During my evaluation when a node in primary rack is down both write and read from the replication rack worked fine.

Can you please share the expected behavior from Dynomite for the cases attached? 

<img width="995" alt="screen shot 2016-10-18 at 11 40 07 am" src="https://cloud.githubusercontent.com/assets/6812846/19491917/94d788f0-9529-11e6-9e91-bf4408791c7e.png">

**Comment by ipapapa** [comment]
The table is not correct. Dyno sees errors and it does not matter if they come from Dynomite or Redis. Dyno will failover for both reads/writes if Dynomite and/or Redis are not available in the local rack. It will failover after 10 attempts.

**Comment by balajivenki** [comment]
@ipapapa  I tested with 1.5.0 and its the same behavior. So I might be wrong. I do not recall the POC dyno version I used. However even with the behavior explained by you when I bring down the primary node in local rack the write after that goes to the node in replication rack. Which is good. But read always fails. Even after 10 attempts. 

In short, when primary node is down write to node in replication rack is working but not the read even after 10 attempts.
